### PR TITLE
feat(card-ui): card flow shipping address screen

### PR DESCRIPTION
**Pull request by esaugomez31**
# Card Shipping Address Screen Implementation

## Overview

Added a new `CardShippingAddressScreen` for the card flow that allows users to view and edit their shipping address. The screen includes editable text fields, dropdown selectors for State/Country, and an informational card with important delivery notes.

### New Screens

**`card-shipping-address-screen.tsx`** - Main shipping address screen with:
- Editable fields for Full name, Address line 1, Address line 2
- Grid layout for City/State and Postal code/Country
- State and Country open a generic selection screen
- InfoCard with bullet points for delivery information

**`selection-screen.tsx`** - Generic reusable selection screen (like language settings) with:
- Receives options via route params
- Uses MenuSelect/MenuSelectItem components
- Callback function for selection handling

### New Components

**`input-field.tsx`** - Renamed and enhanced from `read-only-field.tsx`:
- Added `rightIcon` prop for GaloyIcon (pencil for edit)
- Added `rightIonicon` prop for Ionicons (chevron-down for dropdowns)
- Added `onPress` prop for clickable fields with TouchableOpacity
- Added `ValueStyle` const enum (`Bold` | `Regular`) for different text styles
- Proper type safety with `ValueStyleType`

**`edit-field-modal.tsx`** - Modal for editing text fields:
- Uses CustomModal component
- Save button disabled until value changes
- Trims whitespace from saved values
- Resets to initial value when reopened

### Component Updates

**`info-card.tsx`** - Added bullet list support:
- New optional `bulletItems?: string[]` prop
- Renders bullet points below the description
- Same styling as BulletListCard component

**`multi-line-field.tsx`** - Added press feedback:
- Changed from Pressable to TouchableOpacity
- Added `activeOpacity={0.7}` for consistent press feedback

### Navigation

- Registered `cardShippingAddressScreen` in `root-navigator.tsx`
- Registered `selectionScreen` in `root-navigator.tsx`
- Added typed params in `stack-param-lists.ts` for both screens

### Translations

Added `ShippingAddress` section translations in all languages

### Tests

Created comprehensive tests following project conventions:
- **`edit-field-modal.spec.tsx`** - 23 tests (rendering, button states, interactions, edge cases, rerender)
- **`input-field.spec.tsx`** - Tests for the renamed/enhanced component
- **`selection-screen.spec.tsx`** - 16 tests (rendering, state/country options, interactions, edge cases)
- **`info-card.spec.tsx`** - Added 7 tests for bulletItems feature
- Updated `card-personal-details.spec.tsx` for InputField rename

## Usage Example

```typescript
// Navigate to shipping address screen
navigation.navigate("cardShippingAddressScreen")

// Navigate to selection screen for State
navigation.navigate("selectionScreen", {
  title: LL.CardFlow.ShippingAddress.state(),
  options: US_STATES,
  selectedValue: address.state,
  onSelect: (value: string) => {
    setAddress((prev) => ({ ...prev, state: value }))
    navigation.goBack()
  },
})
```

### CODE REVIEW

File: app/components/card-screen/input-field.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const styles = useStyles({ valueStyle })
+  const {
+    theme: { colors },
+  } = useTheme()
+
+  const rightIconElement = rightIonicon ? (
+    <Icon name={rightIonicon} type="ionicon" size={20} color={colors.primary} />
+  ) : rightIcon ? (
+    <GaloyIcon name={rightIcon} size={20} color={colors.primary} />
+  ) : null
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/card-screen/input-field.tsx:46]: Done!

```suggestion
type IconProps =
  | { rightIcon?: IconNamesType; rightIonicon?: never }
  | { rightIcon?: never; rightIonicon?: string }

type InputFieldProps = {
  label: string
  value: string
  onPress?: () => void
  valueStyle?: ValueStyleType
} & IconProps
```

File: app/components/card-screen/input-field.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  const content = (
+    <View style={styles.valueContainer}>
+      <Text style={styles.value}>{value}</Text>
+      {rightIconElement}
+    </View>
+  )
+
+  const pressableContent = onPress ? (
+    <TouchableOpacity onPress={onPress} activeOpacity={0.7}>
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/card-screen/input-field.tsx]: Done!

```suggestion
<TouchableOpacity
  onPress={onPress}
  activeOpacity={0.7}
  accessibilityRole="button"
  accessibilityLabel={label}
>
```

File: app/screens/settings-screen/selection-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export const SelectionScreen: React.FC = () => {
+  const route = useRoute<SelectionScreenRouteProp>()
+  const { options, selectedValue, onSelect } = route.params
+
+  const [currentValue, setCurrentValue] = React.useState(selectedValue)
+
+  const handleSelect = async (value: string) => {
+    setCurrentValue(value)
+    onSelect(value)
+  }
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/settings-screen/selection-screen.tsx:20]: The `async` keyword is required because `MenuSelect.onChange` expects `Promise<void>` as return type

File: app/i18n/raw-i18n/translations/lg.json [hunk 4]
```diff
@@ -2588,6 +2588,7 @@
         "rate": "Tuwe ekipimo",
         "reauth": "Olutuula lwo luweddeko. Tukusaba oddemu oyingire.",
         "restart": "Ddamu okutandika",
+        "save": "Save",
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/i18n/raw-i18n/translations/lg.json]: Done!

```suggestion
        "save": "Tereka"
```

File: app/i18n/raw-i18n/translations/qu.json [hunk 5]
```diff
@@ -2493,6 +2493,7 @@
         "rate": "Ratiy",
         "reauth": "Ñuqaqa kawsaymi ñawpaqmi kani. Ñuqaqa chaskisqa kaniyta kaniy.",
         "restart": "Musuq kani",
+        "save": "Save",
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/i18n/raw-i18n/translations/qu.json]: Done!

```suggestion
        "save": "Waqaychay"
```

File: app/screens/card-screen/card-shipping-address-screen.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+      case "fullName":
+        return LL.CardFlow.ShippingAddress.fullName()
+      case "addressLine1":
+        return LL.CardFlow.ShippingAddress.addressLine1()
+      case "addressLine2":
+        return LL.CardFlow.ShippingAddress.addressLine2()
+      default:
+        return ""
+    }
+  }
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/card-screen/card-shipping-address-screen.tsx]: Done! Created a narrower type `TextEditableField` that only includes the 3 fields editable via modal

File: app/components/card-screen/edit-field-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      isVisible={isVisible}
+      toggleModal={toggleModal}
+      headerTitle={fieldName}
+      body={
+        <View style={styles.inputContainer}>
+          <TextInput
+            style={styles.input}
+            value={value}
+            onChangeText={setValue}
+            autoFocus
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/card-screen/edit-field-modal.tsx:58]: Done!

```suggestion
            autoFocus
            placeholderTextColor={colors.grey2}
            selectionColor={colors.primary}
            accessibilityLabel={fieldName}
            returnKeyType="done"
            onSubmitEditing={handleSave}
```

File: app/components/card-screen/info-card.tsx [hunk 8]
```diff
@@ -36,13 +38,28 @@ export const InfoCard: React.FC<InfoCardProps> = ({
     return <GaloyIcon name={icon} size={16} color={resolvedIconColor} />
   }
 
+  const renderBulletItems = () => {
+    if (!bulletItems || bulletItems.length === 0) return null
+    return (
+      <View style={styles.bulletList}>
+        {bulletItems.map((item, index) => (
+          <View key={index} style={styles.bulletItem}>
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/components/card-screen/info-card.tsx]: Done!

```suggestion
        {bulletItems.map((item) => (
          <View key={item} style={styles.bulletItem}>
```

File: app/navigation/stack-param-lists.ts [hunk 9]
```diff
@@ -148,6 +148,13 @@ export type RootStackParamList = {
     iconName: IconNamesType
     iconColor?: string
   }
+  cardShippingAddressScreen: undefined
+  selectionScreen: {
+    title: string
+    options: Array<{ value: string; label: string }>
+    selectedValue: string
+    onSelect: (value: string) => void
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/navigation/stack-param-lists.ts:156]: This is intentionally temporary to have a reusable component while working with mock data. Once the real API data is integrated, we'll evaluate using a query-based approach that eliminates the need for this callback parameter

File: app/screens/card-screen/card-shipping-address-screen.tsx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+          </View>
+        </View>
+
+        <View style={styles.gridRow}>
+          <View style={styles.gridItem}>
+            <InputField
+              label={LL.CardFlow.ShippingAddress.postalCode()}
+              value={address.postalCode}
+              valueStyle={ValueStyle.Regular}
+            />
```
[the remark this answers is not in the document]
Comment (esaugomez31) [reply app/screens/card-screen/card-shipping-address-screen.tsx:149]: These fields are intentionally read-only as per the design mockup requirements

**Comment by grimen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3610?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3619** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3614** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3614?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3611** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3611?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3610** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3610?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3610?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3609** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3609?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3608** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3608?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3607** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3607?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3606** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3606?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3605** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3605?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3602** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3601** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3600** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3600?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3598** <a href="https://app.graphite.com/github/pr/blinkbitcoin/blink-mobile/3598?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by grimen — APPROVED** [review]
